src/dmd/lexer.d

```diff
@@ -1064,16 +1064,25 @@ class Lexer : ErrorHandler
                     p++;
                     Token n;
                     scan(&n);
-                    if (n.value == TOK.identifier && n.ident == Id.line)
+                    if (n.value == TOK.identifier)
                     {
-                        poundLine();
-                        continue;
+                        if (n.ident == Id.line)
+                        {
+                            poundLine();
+                            continue;
+                        }
+                        else
+                        {
+                            const locx = loc();
+                            warning(locx, "C preprocessor directive `#%s` is not supported", n.ident.toChars());
+                        }
                     }
-                    else
+                    else if (n.value == TOK.if_)
```

Comment:
Should `#pragma` have special handling here also?

Reply:
I don't see a point to it.